UI/CMakeLists.txt

```diff
@@ -408,6 +408,11 @@ if (APPLE)
 endif()
 set_target_properties(obs PROPERTIES FOLDER "frontend")
 
+if (ENABLE_WAYLAND)
+	find_package(Qt5Gui)
+	include_directories(${Qt5Gui_PRIVATE_INCLUDE_DIRS})
```

Comment:
It's probably not a good idea to rely on Qt's private header files, since there is guarauntee that the headers exist in future version of Qt. What functions are necessary from the private header files? From what I can find in the Qt [docs](https://doc.qt.io/qt-5/qtwaylandcompositor-module.html) regarding wayland compositors, it doesn't seem there is a need for the private headers. Am I missing something?

Reply:
The docs you linked are for Wayland compositors, and not particularly applicable here. We require compositing OBS textures onto the Qt window in an efficient way. So in X11 we get the X11 window and compositing directly into that surface. In wayland they broke this functionality and only offer equivalent access via the private apis. Qt developers have so far stated they have no intention of fixing this in the Qt 5 series.